pkg/buses/reconciler.go

```diff
@@ -63,6 +63,7 @@ const (
 // Subscribe/Unsubscribe happen.
 type Reconciler struct {
 	bus     channelsv1alpha1.GenericBus
+	ref     BusReference
```

Comment:
Not suggesting you change it in this PR, just FYI: `BusReference` is identical to [`types.NamespacedName`](https://godoc.org/k8s.io/apimachinery/pkg/types#NamespacedName). If you expect `BusReference` to evolve independently, it might make sense to keep it.

Reply:
I don't expect BusReference to evolve, but I want to distinguish between BusReference, ChannelReference and SubscriptionReferences in a type safe way. The other significant difference is that the BusReference also handles ClusterBus, which is not namespaced.

Reply:
Makes sense.